clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -200,7 +200,7 @@ public synchronized boolean containsTopic(String topic) {
      * Updates the cluster metadata. If topic expiry is enabled, expiry time
      * is set for topics if required and expired topics are removed from the metadata.
      */
-    public synchronized void update(Cluster cluster, long now) {
+    public synchronized void update(Cluster cluster, boolean hasUnavailablePartitions, long now) {
```

Comment:
@ijuma Thank you for the review. Actually I had initially just passed in the unavailable partitions rather than a boolean. But then found that there were so many references to update() which need to do `Collections.<TopicPartition>emptySet()`. So I did a search-and-replace to use a boolean instead. Now that you have brought it up, perhaps collection was better anyway! I will update the code and add javadoc.